src/RA_Dlg_AchEditor.cpp

```diff
@@ -246,25 +251,39 @@ LRESULT CALLBACK EditProc(HWND hwnd, UINT nMsg, WPARAM wParam, LPARAM lParam) no
             lvDispinfo.item.iSubItem = nSelSubItem;
             lvDispinfo.item.pszText = nullptr;
 
-            TCHAR sEditText[12];
-            GetWindowText(hwnd, sEditText, 12);
-            lvDispinfo.item.pszText = sEditText;
-            lvDispinfo.item.cchTextMax = lstrlen(sEditText);
+            ra::tstring sEditText(12, TCHAR());
+            GetWindowText(hwnd, sEditText.data(), 12);
+            sEditText.shrink_to_fit();
+            lvDispinfo.item.pszText = sEditText.data();
+            lvDispinfo.item.cchTextMax = gsl::narrow<int>(sEditText.length());
 
             HWND hList = GetDlgItem(g_AchievementEditorDialog.GetHWND(), IDC_RA_LBX_CONDITIONS);
 
             // the LV ID and the LVs Parent window's HWND
             SendMessage(GetParent(hList), WM_NOTIFY, static_cast<WPARAM>(IDC_RA_LBX_CONDITIONS),
                         reinterpret_cast<LPARAM>(&lvDispinfo)); // ##reinterpret? ##SD
 
-            // Extra validation?
-
-            if (lstrcmp(sEditText, TEXT("Value")))
+            // Extra validation? - Yes
+            // This function doesn't seem to do what it's supposed to do
+            if (lstrcmp(sEditText.c_str(), TEXT("Value")))
             {
                 // Remove the associated 'size' entry
-                strcpy_s(g_AchievementEditorDialog.LbxDataAt(lvDispinfo.item.iItem,
-                                                             ra::itoe<CondSubItems>(lvDispinfo.item.iSubItem + 1)),
-                         32, "");
+                for (const auto& c : sEditText)
+                {
+                    if (c == char())
+                        break;
+                    if (!std::isdigit(c))
```

Comment:
Pretty sure "0x1234" is a valid value.

Reply:
I'll put "x" as valid. This is to prevent an invalid argument exception